api/v1/nificluster_types.go

```diff
@@ -46,12 +46,18 @@ type NifiClusterSpec struct {
 	Service ServicePolicy `json:"service,omitempty"`
 	// Pod defines the policy for pods owned by NiFiKop operator.
 	Pod PodPolicy `json:"pod,omitempty"`
+	// clusterManager specifies which manager will handle the cluster election and states.
+	// +kubebuilder:default:=zookeeper
+	// +optional
+	ClusterManager ClusterManagerType `json:"clusterManager,omitempty"`
 	// zKAddress specifies the ZooKeeper connection string
 	// in the form hostname:port where host and port are those of a Zookeeper server.
 	// TODO: rework for nice zookeeper connect string =
+	// +optional
```

Comment:
What do you think should change? Because at the end, the Kubernetes management needs no configuration from the user other than to indicate that they want to use it. So in fact, only the ZK configuration becomes optional, no? Yes, sure if the user doesn't provide any ZK configuration it will not work, but the state of the NiFi cluster will indicate it, no?

Reply:
I suppose that is the case. I thought we had discussed needing to create a new CRD version, but no worries.